src/query/service/src/sql/planner/semantic/type_check.rs

```diff
@@ -929,7 +930,12 @@ impl<'a> TypeChecker<'a> {
         let mut arg_types = vec![];
 
         for argument in arguments {
-            let box (arg, arg_type) = self.resolve(argument, None).await?;
+            let box (arg, mut arg_type) = self.resolve(argument, None).await?;
+            if let Scalar::SubqueryExpr(subquery) = &arg {
```

Comment:
Can we `if let Scalar::SubqueryExpr(subquery) = &arg && subquery.typ == SubqueryType::Scalar && !arg.data_type().is_nullable()`?

Reply:
No, `subquery` can't be know in `subquery.typ == SubqueryType::Scalar`